crates/resolver/src/resolver.rs

```diff
@@ -961,7 +1095,17 @@ pub mod testing {
 #[cfg(feature = "tokio-runtime")]
 #[allow(clippy::extra_unused_type_parameters)]
 mod tests {
+    use std::net::{IpAddr, Ipv4Addr};
+    use std::sync::Mutex;
+
     use crate::proto::xfer::DnsRequest;
+    use futures_executor::block_on;
```

Comment:
Nit: keep the `crate::proto` import with the existing `crate::` imports, below (please remove the empty line between the `crate` and `super` imports), and rewrite the `hickory_proto` imports to be `crate::proto` instead.